### PR TITLE
Add #1056 Add a view of contacts who do not have tags associated with them [In review]

### LINKED ISSUE #1056: Add a view of contacts who do not have tags associated with them
Feedback from one of our user:

Would you consider adding a "untagged" view of contacts, so we can see the list of contacts we haven't added tags to yet?

### PR DESCRIPTION
**Pull request by turtles2**
Closes #1056 by adding a untagged tag. Also it adds a method for selecting contacts with an array of tags.

### CODE REVIEW

File: resources/views/people/index.blade.php [hunk 1]
```diff
@@ -42,6 +42,16 @@
                         <a href="/people">{{ trans('people.people_list_clear_filter') }}</a>
                       </p>
                   @endif
+                  @if ($tagLess)
+                      <p class="clear-filter">
+                        {!! trans('people.people_list_filter_tag') !!}
```
Comment (asbiin) [inline resources/views/people/index.blade.php]: The syntax `{!!` is for unescaped echo. Perhaps it's better here to use `{{` for html escape.

File: resources/views/people/index.blade.php [hunk 2]
```diff
@@ -42,6 +42,16 @@
                         <a href="/people">{{ trans('people.people_list_clear_filter') }}</a>
                       </p>
                   @endif
+                  @if ($tagLess)
+                      <p class="clear-filter">
+                        { trans('people.people_list_filter_tag') }
```
Comment (asbiin) [inline resources/views/people/index.blade.php]: The right syntax here is `{{ trans('people.people_list_filter_tag') }}`

File: resources/views/people/index.blade.php [hunk 3]
```diff
@@ -42,6 +42,16 @@
...
+                            <span class="pretty-tag">
+                                { trans('people.people_list_untagged') }
```
Comment (asbiin) [inline resources/views/people/index.blade.php]: Same problem here

File: resources/views/people/index.blade.php [hunk 4]
```diff
@@ -72,6 +82,15 @@
                     <a href="/people">{{ trans('people.people_list_clear_filter') }}</a>
                   </p>
               @endif
+              @if ($tagLess)
+                  <p class="clear-filter">
+                    {{ trans('people.people_list_filter_tag') }}
+                        <span class="pretty-tag">
+                        { trans('people.people_list_untagged') }
```
Comment (asbiin) [inline resources/views/people/index.blade.php]: Same here

File: resources/views/people/index.blade.php [hunk 5]
```diff
@@ -156,6 +175,10 @@
               @include('partials.components.people-upgrade-sidebar')
 
               <ul>
+                  <li>
+                      <span class="pretty-tag"><a href="/people?no_tag=true">{ trans('people.people_list_untagged') }/a></span>
```
Comment (asbiin) [inline resources/views/people/index.blade.php]: Same here

**Comment by djaiss** [comment]
Thanks for this!

However, the layout is somehow broken at this stage.



Overall I like the feature, but I don't really like the implementation. If you are ok with it, I'm going to submit a design for this. Would it be alright?

**Comment by turtles2** [comment]
I am not attached to the desgin. Do you have a problem with the way I
implemented it on the controller and model side?

On Apr 17, 2018 10:35 PM, "Régis Freyd" <notifications@github.com> wrote:

Thanks for this!

However, the layout is somehow broken at this stage.

<https://user-images.githubusercontent.com/61099/38910481-85a73566-4297-11e8-846a-84ec202f02b4.png>

Overall I like the feature, but I don't really like the implementation. If
you are ok with it, I'm going to submit a design for this. Would it be
alright?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/monicahq/monica/pull/1112#issuecomment-382245585>, or mute
the thread

**Comment by turtles2** [comment]
@djaiss Where do we stand with this? I would like to know what direction you want to take this?

**Comment by djaiss** [comment]
@turtles2 Of course I completely forgot :-(

Don't hesitate to ping me because I tend to get carried away with other features. I'll check this asap.

File: app/Contact.php [hunk 6]
```diff
@@ -1291,6 +1291,23 @@ public function getRelatedRealContact()
         }
     }
 
+    public function scopeTags($query, $tags)
```
Comment (djaiss) [inline app/Contact.php]: Please add a comment for this class that briefly explains what it does.

**Comment by djaiss** [comment]
I just wrote a comment, and I think we would be good to go (if you fix the failing tests as well).
I will merge the PR then, and I will change the design in a subsequent PR myself, it'll be quicker that way and I like having complete control of the HTML/CSS :-D

**Comment by turtles2** [comment]
@djaiss The changes have been made and tests are passing this should be good to merge

**Comment by asbiin** [comment]
What if a contact is tagged with the tag "Untagged" ?
image
IMHO it would be better if the (real) "Untagged" string had a different style than the real tags.

**Comment by turtles2** [comment]
I agree. If you have an idea lets do it.

**Comment by asbiin** [comment]
@djaiss Master, what do you think

**Comment by turtles2** [comment]
@djaiss We should be good to merge this in.

**Comment by turtles2** [comment]
@djaiss The Changes have been made. As for the Commits that will be resolved in all PRs going forward.

**Comment by djaiss** [comment]
Thanks @turtles2 😀

**Pull request merged**